apps/cowswap-frontend/src/modules/permit/hooks/useAccountAgnosticPermitHookData.ts

```diff
@@ -43,12 +41,15 @@ function useGeneratePermitHookParams(): GeneratePermitHookParams | undefined {
 
   const permitInfo = usePermitInfo(inputCurrency, tradeType)
 
-  return useSafeMemo(() => {
-    if (!inputCurrency || !('address' in inputCurrency) || !isSupportedPermitInfo(permitInfo)) return undefined
+  const address = inputCurrency && 'address' in inputCurrency && inputCurrency.address
```

Comment:
Instead of this, shouldnt we assert `if (isToken(inputCurrency)) {...}`

`Currency = NativeCurrency | Token`

So I assume what you are check-in is that is a token?

Reply:
Fixed: https://github.com/cowprotocol/cowswap/pull/4275/commits/5323a516583dabc5c1b27c59b08dad6cd6aa8ca1

Reply:
Thank you!

Reply:
Awesome!